nvtabular/io.py

```diff
@@ -458,38 +598,28 @@ def _write_thread(self):
             finally:
                 self.queue.task_done()
 
-    def _write_metadata(self):
-        if self.cats is None:
-            return
-        metadata_writer = open(os.path.join(self.out_dir, "metadata.json"), "w")
-        data = {}
-        data["file_stats"] = []
-        for i in range(len(self.data_files)):
-            data["file_stats"].append({"file_name": f"{i}.data", "num_rows": self.num_samples[i]})
-        # cats
-        data["cats"] = []
-        for c in self.cats:
-            data["cats"].append({"col_name": c, "index": self.col_idx[c]})
-        # conts
-        data["conts"] = []
-        for c in self.conts:
-            data["conts"].append({"col_name": c, "index": self.col_idx[c]})
-        # labels
-        data["labels"] = []
-        for c in self.labels:
-            data["labels"].append({"col_name": c, "index": self.col_idx[c]})
+    @classmethod
+    def write_special_metadata(cls, md, fs, out_dir):
+        # Sort metadata by file name and convert list of
+        # tuples to a list of metadata byte-blobs
+        md_list = [m[1] for m in sorted(list(md.items()), key=lambda x: natural_sort_key(x[0]))]
 
-        json.dump(data, metadata_writer)
-        metadata_writer.close()
+        # Aggregate metadata and write _metadata file
+        _write_pq_metadata_file(md_list, fs, out_dir)
+
+    def _close_writers(self):
+        md_dict = {}
+        for writer, fn in zip(self.data_writers, self.data_fns):
+            md_dict[fn] = writer.close(metadata_file_path=fn)
```

Comment:
Is this right? Isn't data_fns just the filename of the parquet file without the path, and if so should we be using self.data_paths instead (and removing self.data_fns entirely)?

Reply:
The path for each column chunk that is written to the global _metadata file must be a relative path.  Since we will write _metadata in the same directory as the data files, we only want to write the file names.